assets/main.js

```diff
@@ -275,6 +318,16 @@ function getRandomScreenshot() {
     }
 
     const random = Math.trunc(Math.random() * unusedScreenshots.length)
+    // Line below is quite expensive, you dereference an element in
+    // unusedScreenshots, then compare it with every element of screenshots
+    // array until they are or there are no more elements, take its index or -1
+    // (invalid index) and then dereference an element from screenshots array
+    // with that index. Which could be solved if it did not involved 2
+    // unnecessary arrays which together have the same contents and also
+    // introduce additional strain on GC every time you get a new screenshot.
+    // Yes, I agree, it solved problem of never ending random generation, which
+    // I instead of solving delayed by shuffling an array, i.e. random'ing it
+    // instead of drawing its elements randomly.
     const index = screenshots.indexOf(unusedScreenshots[random])
```

Comment:
It is probably not worth checking if nothing was found because it would mean that something else is broken and won't solve or proof anything.